### PR TITLE
Banned team antags are no longer allowed to play even if there are no ghosts to replace them.

### LINKED ISSUE #8307: Blood cult ban not working
Round 29103
Rorimor (banned player)

Blood cult banned and still got converted. I was a ghost at the time and there was no prompt to take him over either.

### PR DESCRIPTION
**Pull request by Bibby0110**
# Github documenting your Pull Request

Fixes https://github.com/yogstation13/Yogstation/issues/8307

# Wiki Documentation

# Changelog

:cl:   
bugfix: Antag banned players can no longer play that antag even if there are no ghosts to replace them
/:cl:

### CODE REVIEW

**Comment by SomeguyManperson** [comment]
would this work? Don't want to have the whole block doubled
```
	if(LAZYLEN(candidates))
		var/mob/dead/observer/C = pick(candidates)
	to_chat(owner, "Your mob has been taken over by a ghost! Appeal your job ban if you want to avoid this in the future!")
	message_admins("[key_name_admin(owner)] [C ? "has been replaced by [key_name_admin(C)]" : "was unable to be replaced, and was ghosted"] since they are banned from [job_rank].")
	owner.current.ghostize(0)
	owner.current.key = C?.key
```

File: code/modules/antagonists/_common/antag_datum.dm [hunk 1]
```diff
@@ -100,12 +100,18 @@ GLOBAL_LIST_EMPTY(antagonists)
 	set waitfor = FALSE
 
 	var/list/mob/dead/observer/candidates = pollCandidatesForMob("Do you want to play as a [name]?", "[name]", null, job_rank, 50, owner.current)
+	
+	owner.current.ghostize(0)
+	to_chat(owner, "Your mob has been taken over by a ghost! Appeal your job ban if you want to avoid this in the future!")
+	
+	owner.current.ghostize(0)
+	owner.current.key = null
+
```
Comment (Bibby0110) [inline code/modules/antagonists/_common/antag_datum.dm]:
```suggestion
	var/mob/dead/observer/C
```

File: code/modules/antagonists/_common/antag_datum.dm [hunk 2]
```diff
@@ -100,12 +100,18 @@ GLOBAL_LIST_EMPTY(antagonists)
...
 	var/list/mob/dead/observer/candidates = pollCandidatesForMob("Do you want to play as a [name]?", "[name]", null, job_rank, 50, owner.current)
+	
+	owner.current.ghostize(0)
+	to_chat(owner, "Your mob has been taken over by a ghost! Appeal your job ban if you want to avoid this in the future!")
+	
+	owner.current.ghostize(0)
+	owner.current.key = null
+	
 	if(LAZYLEN(candidates))
 		var/mob/dead/observer/C = pick(candidates)
```
Comment (Bibby0110) [inline code/modules/antagonists/_common/antag_datum.dm]:
```suggestion
		C = pick(candidates)
```

File: code/modules/antagonists/_common/antag_datum.dm [hunk 3]
```diff
@@ -100,12 +100,18 @@ GLOBAL_LIST_EMPTY(antagonists)
 	set waitfor = FALSE
 
 	var/list/mob/dead/observer/candidates = pollCandidatesForMob("Do you want to play as a [name]?", "[name]", null, job_rank, 50, owner.current)
+	
+	owner.current.ghostize(0)
```
Comment (SomeguyManperson) [inline code/modules/antagonists/_common/antag_datum.dm]: don't think this should show up twice

Comment (Bibby0110) [reply code/modules/antagonists/_common/antag_datum.dm]: correct!

**Review by adamsong — APPROVED** [review]
Good change given how few people take ghosts

**Pull request merged**